csi.proto

```diff
@@ -37,6 +37,23 @@ service Controller {
 
   rpc ControllerGetCapabilities (ControllerGetCapabilitiesRequest)
     returns (ControllerGetCapabilitiesResponse) {}  
+
+  // Snapshot can be used as the source to provision a new volume.
+  // CreateVolumeRequest is modified to take an OPTIONAL parameter of
+  // a source snapshot.
+  // Reverting snapshot, where data in the original volume is erased
+  // and replaced with data in the snapshot, is an advanced
+  // functionality not every storage system can support and therefore
+  // is out of scope for this proposal. It should be added as a separate
+  // function (not in CreateVolume) when proposed in the future.
+  rpc CreateSnapshot (CreateSnapshotRequest)
+    returns (CreateSnapshotResponse) {}
+
+  rpc DeleteSnapshot (DeleteSnapshotRequest)
+    returns (DeleteSnapshotResponse) {}
+
+  rpc ListSnapshots (ListSnapshotsRequest)
```

Comment:
Was there some feedback that a `ListSnapshots` rpc was needed? I'm not sure when an orchestrator would use this.

Reply:
A key workflow that this should support is creating a volume from a snapshot.  I believe that the idea here is that a CO would ask for the list of snapshots so that the user can pick one to clone from.